### PR TITLE
DDF-5610 Automatically convert WKT strings into the correct format

### PR DESCRIPTION
**Pull request by hayleynorton**
##### ABBREVIATED REVIEW
Link to 2.19.x PR: https://github.com/codice/ddf/pull/5611 
___________________________________________________
#### What does this PR do?
Converts WKT strings to the correct format automatically instead of relying on error messages to inform the user of the correct format and making them convert it themselves.
#### Who is reviewing it? 
@zta6 @cassandrabailey293 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@Bdthomson 
#### How should this be tested?
Copy and paste WKT strings into the Polygon and Line geo filter fields. Verify that they're being immediately converted into the correct format, the shapes are still showing up on the map and you can successfully execute the search without any error messages.

Note about multipolygons: each polygon in a multipolygon must have at least 4 coordinate pairs (first and last should be the same). Test with multipolygons where at least one polygon isn't valid and verify that the invalid polygon is simply removed from the multipolygon; the rest of the shapes should still convert and be visible on the map. Also test with a multipolygon where none of the polygons are valid. No conversion should occur and you should still get an error.

Example multipolygon: 
`MULTIPOLYGON (((45.574810133283215 20.62558894559107, 45.5194086 20.6255882, 45.51939284236649 20.66286978112815, 45.574810133283215 20.62558894559107)), ((45.59837085736602 20.762588148838585, 45.64061788382024 20.762594644045915, 45.69038779816324 20.72685765983166, 45.69032921329696 20.649303862907075, 45.613123 20.715271, 45.635361148168556 20.715827143471095, 45.59837085736602 20.762588148838585)))`

Example polygon: 
`POLYGON ((-45.69035002266293 -20.676851, -45.6903113 20.6255905, 45.623261 -20.625589597640413, 45.623261 20.676851, 45.69035002266293 20.676851))`

Example line:
`LINESTRING (45.4806 20.644323,45.620127 20.727819)`
#### What are the relevant tickets?
Fixes: #5610
G-7587 
#### Screenshots
Before:
before
After:
after1-min
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by hayleynorton** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js [hunk 1]
```diff
@@ -108,28 +118,48 @@ class BaseLine extends React.Component {
       return false
     }
   }
+  validatePoint(point) {
+    if (
+      point.length !== 2 ||
+      (Number.isNaN(Number.parseFloat(point[0])) &&
```
Comment (jrnorth) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: Should this be an `||`?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js [hunk 2]
```diff
@@ -150,6 +180,35 @@ class BaseLine extends React.Component {
...
+      .map(shape => shape.match(coordinatePairRegex))
+      .filter(shape => shape !== null && shape.length >= numPoints)
+      .map(shape =>
+        shape.map(coordinatePair => coordinatePair.replace(' ', ','))
+      )
+    return shapes.length === 0
+      ? value
+      : shapes.length === 1
+        ? `[[${shapes[0].join('],[')}]]`
+        : `[${shapes.map(shapeCoords => `[[${shapeCoords.join('],[')}]]`)}]`
```
Comment (jrnorth) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: ✏️ I found this tough to read and understand.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js (hunk 1 above)
Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: Yes I believe you're right

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js (hunk 2 above)
Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: I refactored this slightly, hopefully it's more straightforward now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js [hunk 3]
```diff
@@ -108,28 +118,45 @@ class BaseLine extends React.Component {
...
       } else {
-        if (
-          point[0] > 180 ||
-          point[0] < -180 ||
-          point[1] > 90 ||
-          point[1] < -90
-        ) {
-          message = JSON.stringify(point) + ' is not a valid point.'
+        if (this.props.mode.includes('multi')) {
+          message = `Switch to ${isLine ? 'Line' : 'Polygon'}`
```
Comment (jrnorth) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: ✏️ I'm not sure what this message is for but I find the ternary a bit confusing. If it's currently a line (`isLine` is true) then I'd think the message would be `Switch to Polygon`.

**Comment by bdeining** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js (hunk 3 above)
Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js]: Oh I can see how this is confusing with the naming of the variable. This is here for when/if there's capability to choose multi modes, like `MultiPolygon` or `MultiLine` and you have a two-shape wkt where one of the shapes is invalid (has too few points to be a complete shape). The invalid shape is filtered out in lines 148-154 and the overall string is corrected to a valid individual shape. That's why it informs you to switch to the correct single-shape mode, so that the shape and the mode match. This is definitely a corner case that won't happen often, but needed to be addressed so that the user isn't shown an unrelated default error message

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by hayleynorton** [comment]
build now

**Comment by andrewzimmer** [comment]
Hero Successful 🎉 
- verified WKT values are converted to the correct format and appear on the map
